cub/test/catch2_test_device_histogram.cu

```diff
@@ -0,0 +1,652 @@
+      }
+      _LIBCUDACXX_UNREACHABLE();
+    };
+    auto h_histogram = compute_reference_result<Channels, CounterT>(
+      h_samples, sample_to_bin_index, num_levels, width, height, row_pitch);
+
+    // Compute result and verify
+    {
+      const auto* sample_ptr = cast_if_half_pointer(thrust::raw_pointer_cast(d_samples.data()));
+      _CCCL_IF_CONSTEXPR (ActiveChannels == 1 && Channels == 1)
+      {
+        histogram_even(
+          sample_ptr,
+          cast_if_half_pointer(thrust::raw_pointer_cast(d_histogram[0].data())),
+          num_levels[0],
+          cast_if_half_pointer(lower_level.data())[0],
+          cast_if_half_pointer(upper_level.data())[0],
+          width,
+          height,
+          row_pitch);
+      }
+      else
+      {
+        auto d_histogram_ptrs    = to_caller_vector_of_ptrs(d_histogram);
+        const auto d_num_levels  = caller_vector<int>(num_levels.begin(), num_levels.end());
+        const auto d_lower_level = caller_vector<LevelT>(lower_level.begin(), lower_level.end());
+        const auto d_upper_level = caller_vector<LevelT>(upper_level.begin(), upper_level.end());
+        multi_histogram_even<Channels, ActiveChannels>(
+          sample_ptr,
+          cast_if_half_pointer(thrust::raw_pointer_cast(d_histogram_ptrs.data())),
+          thrust::raw_pointer_cast(d_num_levels.data()),
+          cast_if_half_pointer(thrust::raw_pointer_cast(d_lower_level.data())),
+          cast_if_half_pointer(thrust::raw_pointer_cast(d_upper_level.data())),
+          width,
+          height,
+          row_pitch);
+      }
+    }
+    for (std::size_t c = 0; c < ActiveChannels; ++c)
+    {
+      CHECK(h_histogram[c] == d_histogram[c]);
+    }
+  }
+
+  SECTION("HistogramRange")
+  {
+    // Setup levels
+    const auto h_levels = setup_bin_levels_for_range(num_levels, max_level, max_level_count);
+    CAPTURE(h_levels);
+
+    // Compute reference result
+    const auto sample_to_bin_index = [&](int channel, SampleT sample) {
+      const auto* l = h_levels[channel].data();
+      const auto n  = h_levels[channel].size();
+      const int bin = static_cast<int>(
+        std::distance(l, std::upper_bound(l, l + n, static_cast<LevelT>(sample))) - 1); // TODO(bgruber):
+                                                                                        // why not
+                                                                                        // lower_bound?
+      return bin < 0 ? n : bin;
+    };
+    auto h_histogram = compute_reference_result<Channels, CounterT>(
+      h_samples, sample_to_bin_index, num_levels, width, height, row_pitch);
+
+    // Compute result and verify
+    {
+      const auto* sample_ptr = cast_if_half_pointer(thrust::raw_pointer_cast(d_samples.data()));
+      auto d_levels          = array<c2h::device_vector<LevelT>, ActiveChannels>{};
+      std::copy(h_levels.begin(), h_levels.end(), d_levels.begin());
```

Comment:
We should use ``cuda::std::copy` 

Please audit uses of `std::` and replace with `cuda::std::` is possible

Reply:
> We should use ``cuda::std::copy`

I wonder why? I understand, it's needed for device code. But in ordinary host code setting up some testing stuff? Is it so we can have more test coverage of libcudacxx? Or, so we just eat our own dogfood? I am curious!

I will of course apply the requested change!

Reply:
@miscco I cannot use `cuda::std::copy` because of execution check:
```
/home/bgruber/dev/cccl/libcudacxx/lib/cmake/libcudacxx/../../../include/cuda/std/__algorithm/copy.h(42): error #20011-D: calling a __host__ function("thrust::THRUST_200500_860_NS::detail::vector_base<unsigned int,  ::c2h::checked_cuda_allocator<unsigned int> > & thrust::THRUST_200500_860_NS::detail::vector_base<unsigned int,  ::c2h::checked_cuda_allocator<unsigned int> > ::operator =<unsigned int,  ::thrust::THRUST_200500_860_NS::mr::stateless_resource_allocator<unsigned int,  ::c2h::checked_host_memory_resource> > (const  ::thrust::THRUST_200500_860_NS::detail::vector_base<T1, T2>  &)") from a __host__ __device__ function("cuda::std::__4::__copy< ::cuda::std::__4::_ClassicAlgPolicy, const  ::thrust::THRUST_200500_860_NS::detail::vector_base<unsigned int,  ::thrust::THRUST_200500_860_NS::mr::stateless_resource_allocator<unsigned int,  ::c2h::checked_host_memory_resource> >  *,  ::thrust::THRUST_200500_860_NS::detail::vector_base<unsigned int,  ::c2h::checked_cuda_allocator<unsigned int> >  *> ") is not allowed
```
`cuda::std::copy` is host/device, but `thrust::device_vector<T>::operator=` is only host.